### ISSUE TITLE
Make Deprecation Warning in matplotlib more useful

### ISSUE THREAD

**Issue by HDembinski** [matplotlib/matplotlib#13352]
### Bug report

Sources of deprecation warnings in matplotlib are hard to track down.

**Code for reproduction**

```python
from matplotlib import pyplot as plt
plt.tick_params(left="False")
```

**Actual outcome**
```
/home/hdembinski/.local/lib/python3.7/site-packages/matplotlib/cbook/__init__.py:424: MatplotlibDeprecationWarning: 
Passing one of 'on', 'true', 'off', 'false' as a boolean is deprecated; use an actual boolean (True/False) instead.
  warn_deprecated("2.2", "Passing one of 'on', 'true', 'off', 'false' as a "
```
As you can see, there is no indication which line in my client code triggers the warning. If the script is long and not minimal, the origin is difficult to find.

**Expected outcome**
The deprecation warning should be enhanced with a backtrace of some sort, that allows me to find the call in my script which triggers the warning.

**Matplotlib version**
  * Matplotlib version: 3.0.2
  * Python version: 3.7.1

**Comment by jklymak** [comment]
Thanks.  The deprecation operator was recently improved:

```python
from matplotlib import pyplot as plt
fig, ax = plt.subplots()
print(ax.aname)
```

returns the (hopefully) more helpful:

```
testdep.py:3: MatplotlibDeprecationWarning:
The aname attribute was deprecated in Matplotlib 3.1 and will be removed in 3.3.
  print(ax.aname)
Axes
```

This change came in #12152 and will be available when 3.1 is released (Feb/Mar timeframe).

I'll close as fixed.

**Comment by HDembinski** [comment]
Cool, thanks!
